File: Euclid/Messages/Outgoing/ADDITEM.cs
using Euclid.Game;
using Euclid.Network.Streams.Util;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Euclid.Messages.Outgoing
{
    class ADDITEM : IMessageComposer
    {
        private Item item;

        public ADDITEM(Item item)
        {
            this.item = item;
        }

        public override void Write()
        {
            this.item.Serialise(Data);
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/ADDITEM.java
package euclid.messages.outgoing;

import euclid.game.item.Item;
import euclid.messages.IMessageComposer;

public class ADDITEM extends IMessageComposer {

    private final Item item;

    public ADDITEM(Item item) {
        this.item = item;
    }

    @Override
    public void write() {
        item.serialise(getData());
    }
}
